fragments/spring-boot-app-sso-auth-code-flow/src/main/java/com/vmware/tap/fragments/appssoauthcodeflow/web/OidcUserController.java

```diff
@@ -0,0 +1,17 @@
+package com.vmware.tap.fragments.appssoauthcodeflow.web;
+
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class OidcUserController {
+
+    @GetMapping("/protected/profile")
+    public String user(Model model, @AuthenticationPrincipal OidcUser authenticatedUser) {
+        model.addAttribute("username", authenticatedUser.getClaims().get("sub"));
```

Comment:
Optional

```suggestion
        model.addAttribute("username", authenticatedUser.getName());
```

The name is pre-populated with the "sub" claim in `OidcUser`.

If you intentionally want to showcase that you can get the claims in the Controller, then you can leave it as is.